### PR TITLE
Djm/matrix delays

### PR DESCRIPTION
**Pull request by dajmcdon**

### CODE REVIEW

**Review by jiapivialiu — CHANGES_REQUESTED** [review]
They all look good to me except a typo and a potential bug in delay matrix for time-varying delays. They are very helpful!

File: vignettes/delay-distributions.Rmd [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+necessary code is the following.
+
+```{r backer-matrix, eval=FALSE}
+n <- nrow(cancovid)
+backer_delay <- rev(c(0, delay, rep(0, n - length(delay) - 1)))
+delay_mat <- matrix(0, n, n)
+delay_mat[1,1] <- 1
+for (iter in 2:n) delay_mat[iter, 1:iter] <- backer_delay[1:iter]
+delay_mat <- drop0(as(delay_mat, "CsparseMatrix")) # make it sparse, not necessary
+delay_mat <- delay_mat / rowSums(delay_mat) # renormalize
```
[the remark this answers is not in the document]
Comment (dajmcdon) [reply vignettes/delay-distributions.Rmd:170]: I don't think any are 0...

[the remark this answers is not in the document]
Comment (jiapivialiu) [reply vignettes/delay-distributions.Rmd:172]: `delay` is a nonzero vector of length 15, but `n` is more than 1000 here, so `backer_delay` is filled by 0 at the beginning, which makes more than 1000 rows all 0s.

[the remark this answers is not in the document]
Comment (dajmcdon) [reply vignettes/delay-distributions.Rmd:172]: I see. For the future, it's helpful to explain a little more fully or suggest a change.

File: vignettes/delay-distributions.Rmd [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+with a matrix. This requires a bit of work, but is not too challenging. For 
+example, to create the correct matrix using the Baker et al. delays, the 
+necessary code is the following.
+
+```{r backer-matrix, eval=FALSE}
+n <- nrow(cancovid)
+backer_delay <- rev(c(0, delay, rep(0, n - length(delay) - 1)))
+delay_mat <- matrix(0, n, n)
+delay_mat[1,1] <- 1
+for (iter in 2:n) delay_mat[iter, 1:iter] <- backer_delay[1:iter]
```
Comment (dajmcdon) [inline vignettes/delay-distributions.Rmd:170]:
```suggestion
backer_delay <- c(0, delay, rep(0, n - length(delay) - 1))
delay_mat <- matrix(0, n, n)
delay_mat[1,1] <- 1
for (iter in 2:n) delay_mat[iter, 1:iter] <- rev(backer_delay[1:iter])
```

**Review by jiapivialiu — APPROVED** [review]

**Pull request merged**